pkg/dockerauth/multi.go

```diff
@@ -0,0 +1,32 @@
+package dockerauth
+
+import (
+	"fmt"
+
+	"github.com/fsouza/go-dockerclient"
+)
+
+type multiAuthProvider struct {
+	providers []AuthProvider
+}
+
+func NewMultiAuthProvider(providers ...AuthProvider) *multiAuthProvider {
+	return &multiAuthProvider{
+		providers: providers,
+	}
+}
+
+func (p *multiAuthProvider) GetAuthConfiguration(registry string) (docker.AuthConfiguration, error) {
+	for _, provider := range p.providers {
+		authConf, err := provider.GetAuthConfiguration(registry)
+		if err == nil {
```

Comment:
While I agree on the point of exposing errors, I think "keep trying to the best of our abilities, in priority order, until we run out of options" is the most suitable approach for authentication in general (making the actual backends fail-fast is of course desirable). OpenSSH and PAM comes to mind. The simplest thing would be to just log the errors then and there. Your thoughts?